sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/implementation/ChangeFeedContextClientImpl.java

```diff
@@ -82,15 +84,16 @@ public ChangeFeedContextClientImpl(CosmosAsyncContainer cosmosContainer, Schedul
     }
 
     @Override
-    public Flux<FeedResponse<CosmosItemProperties>> createDocumentChangeFeedQuery(CosmosAsyncContainer collectionLink, ChangeFeedOptions feedOptions) {
+    public Flux<FeedResponse<JsonNode>> createDocumentChangeFeedQuery(CosmosAsyncContainer collectionLink,
+                                                                      ChangeFeedOptions feedOptions) {
         AsyncDocumentClient clientWrapper =
             CosmosBridgeInternal.getAsyncDocumentClient(collectionLink.getDatabase());
-        Flux<FeedResponse<CosmosItemProperties>> feedResponseFlux =
+        Flux<FeedResponse<JsonNode>> feedResponseFlux =
             clientWrapper.queryDocumentChangeFeed(BridgeInternal.extractContainerSelfLink(collectionLink), feedOptions)
                                                                     .map(response -> {
-                                                                        List<CosmosItemProperties> results = response.getResults()
+                                                                        List<JsonNode> results = response.getResults()
                                                                                                                      .stream()
-                                                                                                                     .map(document -> new CosmosItemProperties(document.toJson()))
+                                                                                                                     .map(document -> document.toObject(JsonNode.class))
```

Comment:
does `document.toObject(JsonNode.class)` return a reference to internal propertybag or does it copy data?

Reply:
Should return a reference and doesnt copy data